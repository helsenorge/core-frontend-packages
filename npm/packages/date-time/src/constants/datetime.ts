/* DATEPICKER RESOURCES */
export const CALENDAR_LABEL = 'Kalender';
export const CLOSE_DATEPICKER = 'Lukk';
export const FOCUS_STARTDATE = 'Bruk kalenderen og velg dato.';
export const CLEAR_DATE = 'Fjern dato';
export const CLEAR_DATES = 'Fjern datoer';
export const PICK_MONTH = 'Velg måned';
export const PICK_MONTH_EN = 'Pick month';
export const PICK_MONTH_SE = 'Vállje mánu';
export const PICK_YEAR = 'Velg år';
export const PICK_YEAR_EN = 'Pick year';
export const PICK_YEAR_SE = 'Vállje jagi';
export const JUMP_TO_PREVMONTH = 'Gå bakover til forrige måned';
export const JUMP_TO_NEXTMONTH = 'Gå fremover til neste måned';
export const KEYBOARD_SHORTCUTS = 'Tastatursnarveier';
export const SHOW_KEYBOARD_SHORTCUTS_PANEL = 'Åpne panelet med tastatursnarveier';
export const HIDE_KEYBOARD_SHORTCUTS_PANEL = 'Lukk panelet med tastatursnarveier';
export const ENTER_KEY = 'Enter tast';
export const LEFT_ARROW_RIGHT_ARROW = 'Høyre og venstre piltast';
export const UP_ARROW_DOWN_ARROW = 'Opp og ned piltast';
export const PAGE_UP_PAGE_DOWN = 'Side opp og Side ned tast';
export const HOME_END = 'Hjem og Slutt tast';
export const ESCAPE = 'Escape tast';
export const QUESTION_MARK = 'Spørsmålstegn';
export const OPEN_THIS_PANEL = 'Åpne dette panelet';
export const SELECT_FOCUSED_DATE = 'Velg datoen som har fokus';
export const MOVE_FOCUS_BY_ONE_DAY = 'Flytt en dag bakover (venstre) og fremover (høyre)';
export const MOVE_FOCUS_BY_ONE_WEEK = 'Flytt en uke bakover (opp) og fremover (ned)';
export const MOVE_FOCUS_BY_ONE_MONTH = 'Skifte måned';
export const MOVE_FOCUS_START_END_WEEK = 'Gå til første eller siste dag av uken';
export const RETURN_FOCUS_TO_INPUT = 'Gå tilbake til datoen i inputfeltet';
export const SCREEN_READER_INPUT = 'Skriv inn dato eller tab videre for å bruke kalenderen';
export const DEFAULT_FORMAT_INDICATION_NB = '(dd.mm.åååå)';
export const DEFAULT_FORMAT_INDICATION_EN = '(dd.mm.yyyy)';
export const DEFAULT_FORMAT_INDICATION_SE = '(bb.mm.jjjj)';
export const DEFAULT_DATE_PLACEHOLDER_NB = 'dd.mm.åååå';
export const DEFAULT_DATE_PLACEHOLDER_EN = 'dd.mm.yyyy';
export const DEFAULT_DATE_PLACEHOLDER_SE = 'bb.mm.jjjj';
export const DEFAULT_STARTDATE_PLACEHOLDER = 'Startdato';
export const DEFAULT_ENDDATE_PLACEHOLDER = 'Sluttdato';

/* DATEPICKER ERROR RESOURCES */
export const ERROR_INVALID_DATE = 'Datoen er ugyldig';
export const ERROR_INVALID_DATE_RANGE = 'Perioden som er valgt er ufullstendig eller ugyldig';
export const ERROR_REQUIRED_DATE = 'Datoen er påkrevd';
export const ERROR_REQUIRED_DATE_RANGE = 'Både startdato og sluttdato er påkrevd';
export const ERROR_INVALID_MIN_NIGHTS = 'Perioden er ugyldig. Minimum antall dager er: ';
export const ERROR_DATE_BEFORE_MIN = 'Datoen er før den minste tillatte datoen';
export const ERROR_DATE_AFTER_MAX = 'Datoen er etter den eldste tillatte datoen';

/* DATE-TIME-PICKER ERROR RESOURCES */
export const ERROR_TIME_BEFORE_MIN = 'Tispunktet som er valgt er før det minste tillatte tidspunktet';
export const ERROR_TIME_AFTER_MAX = 'Tispunktet som er valgt er etter det eldste tillatte tidspunktet';

/* TIME-INPUT RESOURCES */
export const DATE_SEPARATOR = '-';
export const TIME_SEPARATOR = ':';
export const PLACEHOLDER_HOURS = 'timer';
export const PLACEHOLDER_MINUTES = 'min';

/* TIME-INPUT ERROR RESOURCES */
export const ERROR_INVALID_TIME = 'Tidspunktet er ugyldig';
export const ERROR_REQUIRED_TIME = 'Tidspunktet er påkrevd';
export const ERROR_HOURS_BEFORE_MIN = 'Antall timer er for lavt';
export const ERROR_HOURS_AFTER_MAX = 'Antall timer overskrider det som er tillatt';
export const ERROR_MINUTES_BEFORE_MIN = 'Antall minutter er for lavt';
export const ERROR_MINUTES_AFTER_MAX = 'Antall minutter overskrider det som er tillatt';

/* YEAR-INPUT ERROR RESOURCES */
export const ERROR_INVALID_YEAR = 'Du må skrive inn et gyldig årstall';
export const ERROR_REQUIRED_YEAR = 'Årstall er påkrevd';
export const ERROR_YEAR_BEFORE_MIN_DATE = 'Årstall er før det minste tillatte år';
export const ERROR_YEAR_AFTER_MAX_DATE = 'Årstall er etter det eldste tillatte år';

/* YEAR-MONTH-INPUT RESOURCES */
export const MONTH_PLACEHOLDER = 'Velg måned';
export const YEAR_PLACEHOLDER = 'Velg år';
export const ERROR_INVALID_YEAR_MONTH = 'Ugyldig verdi';
export const ERROR_INVALID_YEAR_MONTH_YEAR = 'Du må skrive inn et gyldig årstall';
export const ERROR_REQUIRED_YEAR_MONTH = 'Årstall og måned er påkrevd';
export const ERROR_YEAR_MONTH_BEFORE_MIN_DATE = 'Tidspunkt er før minste tillatte tidspunkt';
export const ERROR_YEAR_MONTH_AFTER_MAX_DATE = 'Tidspunkt er etter eldste tillatte tidspunkt';
