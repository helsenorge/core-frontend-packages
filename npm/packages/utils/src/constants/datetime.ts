export default {
  DATE_SEPARATOR: '-',
  TIME_SEPARATOR: ':',
};
