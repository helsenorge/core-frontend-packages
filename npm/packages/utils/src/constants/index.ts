import Languages from './languages';
import Projects from './projects';
import Layout from './layout';

export { Languages, Projects, Layout };
