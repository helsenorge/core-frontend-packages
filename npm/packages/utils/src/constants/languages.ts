export default {
  NORWEGIAN: 'nb-no',
  NORWEGIAN_NYNORSK: 'nn-no',
  ENGLISH: 'en-US',
};
