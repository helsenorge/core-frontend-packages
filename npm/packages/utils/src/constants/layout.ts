/*  This file has been converted to TypeScript
 When making changes to this file please make them
 in the file with a .ts or .tsx extension located in
 the Common/src/toolkit folder structure
 Then run the command "npm run build-toolkit" to generate the jsx file
 Please check both files into TFS */

export default {
  ONE_TO_TWO_COLUMN: '39rem',
  TWO_TO_THREE_COLUMN: '59rem',
  THREE_TO_FOUR_COLUMN: '79rem',
};
