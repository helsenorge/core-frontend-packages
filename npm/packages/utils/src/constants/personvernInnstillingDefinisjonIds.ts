export enum PersonvernInnstillingDefinisjonGuids {
  Bruksvilkar = '7AC2178D-0E89-4251-9920-A496B2969640',
  Pha = '2A132834-A9FF-4FA7-B104-7FF6C1218195',
  Pasientreiser = 'D39D21AD-8BF2-4FD3-B73B-02D21EE116DB',
  Ungdomslosning = '44EEAEA9-B70A-4312-AA62-54E96A8AA802',
  InnsynssperrePasientjournal = 'D34A19D3-A474-4729-B097-DC4FE9515032',
  InnsynMinHelse = '49770ACC-2E1B-4A8C-AC88-6BF9D18BFBCE',
  RepresentasjonAvBarn = '9D6DCF50-92DA-468E-8D8C-0B59FF27904F',
  KprDirekteIdentifiserbare = 'D04A5C44-099D-4197-A322-2F9CB4B985B4',
  KprAndreFormal = 'D53CF0DF-D029-4EF7-BDE9-36ACBD545BF0',
  KprGamle = '2A5CC4D7-AB84-4449-B58A-5937A3F5DBD1',
  Fastlegeordningen = '71FF6942-A738-4C43-8C9B-57D1ED4F785F',
  PllAlleUnntattFastlegen = 'FB922810-01F1-4B20-9999-BB25B47D1B44',
  PllBestemtHelsepersonell = '07A49957-F6D8-4D2E-86E2-FE11997709E0',
  InnsynIOpplysningerRegistrertOmMeg = '6EA1AA04-FAD7-48C8-8432-1ACB3FBF6D7B',
  InnsynIPasientjournal = 'DC5B5D14-DEED-48E5-BF6D-99ED77F08A09',
  DigitalHelsetjeneste = 'FB640456-AE6F-4664-92C8-C6739D76CAEC',
}
