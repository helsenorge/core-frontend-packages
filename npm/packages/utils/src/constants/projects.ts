export default {
  DIALOG: 'HN.MinHelse.Dialog.JS',
  DIALOG_VP: 'HN.MinHelse.Dialog.VP.JS',
  PASIENTJOURNAL: 'HN.MinHelse.Pasientjournal',
  BYTTFASTLEGE: 'HN.MinHelse.ByttFastlege.JS',
  FASTLEGEN: 'HN.MinHelse.DinFastlege.JS',
  COMMON: 'HN.MinHelse.Common.JS',
  PORTAL: 'HN.Portal',
  ARKIV: 'HN.MinHelse.HealthArchive.Strings.JS',
  EGENANDEL: 'HN.MinHelse.HFT.Strings.JS',
  KJERNEJOURNAL: 'HN.MinHelse.KJ.Strings.JS',
  PERSONVERN: 'HN.MinHelse.Personvern.Strings',
  HELSEREGISTRE: 'HN.MinHelse.Helseregistre.JS',
  FORSKNING: 'HN.MinHelse.Forskning.JS',
  RESEPTER: 'HN.MinHelse.Resepter.JS',
  NHN: 'HN.MinHelse.NHN.Strings',
  CONTENT: 'HN.MinHelse.Content',
  SKJEMAUTFYLLER: 'HN.MinHelse.Skjemautfyller',
  VELGBEHANDLINGSSTED: 'HN.VelgBehandlingssted.JS',
  VAKSINER: 'HN.MinHelse.Vaksiner.JS',
  LEGEMIDLER: 'HN.MinHelse.Legemidler.JS',
  SYSVAK: 'HN.MinHelse.Sysvak',
  PASIENTREISER: 'HN.MinHelse.Pasientreiser.JS',
  MFR: 'HN.MinHelse.Mfr',
  DONORKORT: 'HN.MinHelse.Donorkort',
  VERKTOY: 'HN.MinHelse.Verktoy',
  SECURITY: 'HN.MinHelse.SecurityFramework',
};
