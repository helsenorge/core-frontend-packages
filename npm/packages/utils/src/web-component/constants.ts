export enum HeaderFooterEvents {
  setlistenerready = 'hn-webcomp-header-footer-event-setlistenerready',
  setsimplifiedheader = 'hn-webcomp-header-footer-event-setsimplifiedheader',
  setanonymousheader = 'hn-webcomp-header-footer-event-setanonymousheader',
  sethiddenheader = 'hn-webcomp-header-footer-event-sethiddenheader',
  setdriftsmeldingpath = 'hn-webcomp-header-footer-event-setdriftsmeldingpath',
  setsimplifiedfooter = 'hn-webcomp-header-footer-event-setsimplifiedfooter',
  sethiddenfooter = 'hn-webcomp-header-footer-event-sethiddenfooter',
  setvarslinger = 'hn-webcomp-header-footer-event-setvarslinger',
  refreshvarslinger = 'hn-webcomp-header-footer-event-refreshvarslinger',
  refreshvarslingeroghendelsesmeny = 'hn-webcomp-header-footer-event-refreshvarslingeroghendelsesmeny',
  setonshowsignoutbox = 'hn-webcomp-header-footer-event-setonshowsignoutbox',
}

export enum CmsBlocksEvents {
  setHiddenPromopanel = 'hn-webcomp-cms-blocks-event-sethiddenpromopanel',
}

export enum CommonEvents {
  isMounted = 'hn-webcomp-common-ismounted',
}
