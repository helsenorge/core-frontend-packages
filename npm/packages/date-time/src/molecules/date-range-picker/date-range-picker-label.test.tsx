import * as React from 'react';
import { mount } from 'enzyme';
import { LanguageLocales } from '@helsenorge/core-utils/constants/languages';
import DateRangePickerLabel from './date-range-picker-label';

describe('DateRangePicker date-range-picker-label', () => {
  describe('Gitt at DateRangePickerLabel skal rendres', () => {
    describe('Når den bare har default verdier', () => {
      const wrapper = mount(<DateRangePickerLabel label={'myLabel'} isRequired={false} />);
      it('Så rendres den med rikitg css classer og legend (label med eksplisitt format)', () => {
        expect(wrapper.find('legend').prop('className')).toEqual('datepicker__legend');
        expect(wrapper.find('legend').text()).toEqual('myLabel (dd.mm.åååå)');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har isLabelHidden', () => {
      const wrapper = mount(<DateRangePickerLabel label={'myLabel'} isRequired={false} isLabelHidden />);
      it('Så rendres den med rikitg css classer', () => {
        expect(wrapper.find('legend').prop('className')).toEqual('datepicker__legend');

        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den er required og har requiredLabel', () => {
      const wrapper = mount(<DateRangePickerLabel label={'myLabel'} isRequired={true} requiredLabel={'MyRequiredLabel'} />);
      it('Så rendres den med rikitg legend (label med requiredLabel og eksplisitt format)', () => {
        expect(wrapper.find('legend').text()).toEqual('myLabel MyRequiredLabel (dd.mm.åååå)');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har optionalLabel', () => {
      const wrapper = mount(<DateRangePickerLabel label={'myLabel'} isRequired={false} optionalLabel={'MyOptionalLabel'} />);
      it('Så rendres den med rikitg legend (label med optionalLabel og eksplisitt format)', () => {
        expect(wrapper.find('legend').text()).toEqual('myLabel (dd.mm.åååå) MyOptionalLabel');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har english locale', () => {
      const wrapper = mount(<DateRangePickerLabel locale={LanguageLocales.ENGLISH} label={'myLabel'} isRequired={false} />);
      it('Så rendres den med riktig legend (riktig engelsk format)', () => {
        expect(wrapper.find('legend').text()).toEqual('myLabel (dd.mm.yyyy)');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har english locale og optionalLabel', () => {
      const wrapper = mount(
        <DateRangePickerLabel locale={LanguageLocales.ENGLISH} label={'myLabel'} isRequired={false} optionalLabel={'MyOptionalLabel'} />
      );
      it('Så rendres den med riktig legend (riktig engelsk format)', () => {
        expect(wrapper.find('legend').text()).toEqual('myLabel (dd.mm.yyyy) MyOptionalLabel');
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har helpButton', () => {
      const wrapper = mount(
        <DateRangePickerLabel label={'myLabel'} isRequired={false} helpButton={<span className={'helpButton'}>{'helpButton'}</span>} />
      );
      it('Så rendres den med helpButton nederst', () => {
        expect(wrapper.find('.helpButton').length).toEqual(1);
        expect(wrapper.render()).toMatchSnapshot();
      });
    });

    describe('Når den har subLabel', () => {
      const wrapper = mount(
        <DateRangePickerLabel label={'myLabel'} isRequired={false} subLabel={<span className={'subLabel'}>{'subLabel'}</span>} />
      );
      it('Så rendres den med subLabel nederst i legenden', () => {
        expect(wrapper.find('.subLabel').length).toEqual(1);
        expect(wrapper.render()).toMatchSnapshot();
      });
    });
  });
});
