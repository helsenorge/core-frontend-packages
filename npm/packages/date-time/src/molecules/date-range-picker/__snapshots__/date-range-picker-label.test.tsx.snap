// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den bare har default verdier Så rendres den med rikitg css classer og legend (label med eksplisitt format) 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label "
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.åååå)
  </em>
</legend>
`;

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den er required og har requiredLabel Så rendres den med rikitg legend (label med requiredLabel og eksplisitt format) 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label "
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     MyRequiredLabel
  </em>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.åååå)
  </em>
</legend>
`;

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den har english locale Så rendres den med riktig legend (riktig engelsk format) 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label "
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.yyyy)
  </em>
</legend>
`;

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den har english locale og optionalLabel Så rendres den med riktig legend (riktig engelsk format) 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label "
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.yyyy) MyOptionalLabel
  </em>
</legend>
`;

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den har helpButton Så rendres den med helpButton nederst 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label "
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.åååå)
  </em>
  <span
    class="helpButton"
  >
    helpButton
  </span>
</legend>
`;

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den har isLabelHidden Så rendres den med rikitg css classer 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label datepicker__legend__label--hidden"
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.åååå)
  </em>
</legend>
`;

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den har optionalLabel Så rendres den med rikitg legend (label med optionalLabel og eksplisitt format) 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label "
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.åååå) MyOptionalLabel
  </em>
</legend>
`;

exports[`DateRangePicker date-range-picker-label Gitt at DateRangePickerLabel skal rendres Når den har subLabel Så rendres den med subLabel nederst i legenden 1`] = `
<legend
  class="datepicker__legend"
>
  <span
    class="datepicker__legend__label "
  >
    myLabel
  </span>
  <em
    class="datepicker__legend__sublabel"
  >
     (dd.mm.åååå)
  </em>
  <span
    class="label__sublabel"
  >
    <span
      class="subLabel"
    >
      subLabel
    </span>
  </span>
</legend>
`;
